### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by ArkSurfer**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
- # Code to create a new book entry in the database.
-    todo = request.form.get('todoitem')
-    session.add_item(todo)
-    return render_template('index.html', list=session.get_items())
+    client.move_card_to_done(id)
+    return redirect(url_for('index'))
 
+#     cards = todo_list of cards
+#     card = next((x for x in cards if x.id == id), None)   
+#     card.move card to(done_list.id)
```
Comment (hugh-emerson) [inline app.py:40]: This can be deleted.

File: app.py [hunk 2]
```diff
@@ -1,32 +1,43 @@
 from flask import Flask, render_template, request, redirect, url_for
+from dotenv import load_dotenv
+from item import TrelloItem
+from trello import TrelloClient
 import session_items as session
+import os
+import requests
+import json
```
Comment (hugh-emerson) [inline app.py:8]: Unused imports: dotenv, session_items, requests, json

And in fact, now that you have moved away from using session, the files session_items.py and flask_config.py can be deleted, and the app.config line below can be deleted.

File: app.py [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
 
-@app.route('/')
-def index():
+client = TrelloClient(
+    os.getenv('API_KEY'),
+    os.getenv('TOKEN')
```
Comment (hugh-emerson) [inline app.py:15]: You should update .env.template to contain all the environment variables (either blank or with obviously placeholder values)

SECRET_KEY is no longer needed.

And the README needs updating. It should contain everything the reader needs to know to run the app after cloning it.

File: trello.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        return {"key": self.key, "token": self.token}
+
+    def get_all_cards_for_board(self, board_id):
+        response = requests.get(f"https://api.trello.com/1/boards/{board_id}/cards", params=self.get_auth_params())
+        response_json = response.json()
+        return response_json
+
+    def add_card_to_list (self, list_id, card_name):
+        params = self.get_auth_params()
+        params["idList"] = list_id
```
Comment (hugh-emerson) [inline trello.py:19]: Really minor point, but TrelloClient is slightly inconsistent at the moment.
When creating cards, app.py passes in the list id.
When completing cards, TrelloClient looks up the list id with os.getenv.
The key and token are passed in when initialising TrelloClient. 

It'd be nicer to stick to something more consistent

File: templates/index.html [hunk 5]
```diff
@@ -9,15 +9,20 @@ <h1 class="display-4">To-Do App</h1>
...
-              </li> 
+        {% for item in items %}
+          <li class="list-group-item flex-column">
+            <div class="d-flex w-80 justify-content-between">
+              <h5 class="mt-1 mb-1">{{ item.title }}</h5>
+              {% if item.status != 'Done' %}
+                <a href="{{ url_for('complete_item', id=item.id) }}" class="btn btn-primary">Mark Done</a>
+              {% endif %}
+			  {% if item.status == 'Done' %}
+                <a class= "btn btn-primary">Done</a>
```
Comment (hugh-emerson) [inline templates/index.html:22]: Why is this a button styled anchor (`<a>`)? It doesn't do anything

**Review by hugh-emerson — APPROVED** [review]
Looks great! I've left some comments but the only important one is to remember to update the contents of README.md

Also deleting unused code like session_items.py is easy and keeps the codebase significantly tidier.

**Comment by ArkSurfer** [comment]
closed, later branch had changes

**Pull request closed without merging**